conventions/cli.md

```diff
@@ -0,0 +1,56 @@
+# Guidelines for kn Commands
+
+Commands are generally of the form `kn <resource> <verb>`.
+
+## Resource
+
+The <group>.knative.dev Kind, singluar and lowercase. For example, `service` for
+`serving.knative.dev/service` or `trigger` for `eventing.knative.dev/trigger`.
+
+## Verb
+
+If the thing the user's doing can be described by the following commands, these
+should be the name of the verb:
+
+* `describe` prints detailed information about a single resource. It can include
```

Comment:
At the moment we don't have really a `describe` implemented, so it would be nice to reflect on this for a moment (i.e. it's still easy to change)

We had some brief discussion whether we want to rename that to `show` to distinguish it from the `kubectl` case as with `kn service show myservice -o yaml` we will be able to get the resource description of a Knative resource.

We already moved from `get` to `list` so it would be consequent also to move away from `describe` to avoid any confusion. I don't insist on `show` (but its much shorter to type)

| Action | kubectl | kn |     
| ------ | ------- | --- |
| List resources - human readable | kubectl get <entity> |  kn <entity> list |
| List resources - machine readable with -o | list or single entity depending on arguments | always list |
| Resource details - human readable | kubectl describe <entity> | kn <entity> show |
| Resource details - machine readable with -o |  ----    | always single entity |

Reply:
I like "describe", personally. And I like not changing things — describe exists today, it's just got annoying format.

Reply:
I'm fine with keeping "describe" then (as it looks like I'm the only one with this concern)